### DISCUSSION TITLE
E19-433m30s 1 W (smd) kompatybilnosc biblioteki.

### DISCUSSION THREAD

**Discussion by farmaceuta777** [jgromes/RadioLib#238] [Q&A]
Will this library work with the module from the title ??
I was looking for modules with SPI communication on Ali and found only the one with 1watt power ..
second question..
does this library allow the use of 1watt power? ... in the code I only see a possibility to change to 20dbi .. I will be grateful for the answers!
(I am a beginner)

sorry for english ... translator: - /

**Comment by jgromes** [comment]
E19-433M30S is essentially an SX1278 chip with external PA. It has an SPI interface, so it will work with the library.

The 1 W output power doesn't have much to do with SX1278, since the E19 modules adds an external PA to it, and the library can only control the SX1278 internal PA. That one only goes up to 22 dBm. To use this module, you will have to find out what sort of PA it's using, I suggest contacting the manufacturer. Then you will know which sort of power setting to use in the library (to not oversaturate the external PA).

For example, E22-433M30S is a similar LoRa module from Ebyte (based on SX1268). It too has a 1W power output and external PA. This specific module uses 26 dB amplifier, so to get the 1 W power output, you would set SX1278 to only transmit at 4 dBm (since 4 dBm + 26 db = 30 dBm).

**Reply by farmaceuta777** [reply]
Thank you very much for the comprehensive answer! :-)
could you tell me what are the possible power ranges on this module
E22-433m30s?
(I'm a beginner, and for me it's not that easy ..)

**Reply by jgromes** [reply]
Because than one has the 26 dB PA and is using SX1268, the possible output power range is 21 to 30 dBm. Keep in mind that you will have to set the power in library by 26 dB lower because of that external PA.

Also, transmitting on this high power might require license, depending on where you live.

**Reply by farmaceuta777** [reply]
So if I want to set the power to 21dbi I have to use the value '-5' in the library? Thank you for your answer..

**Reply by jgromes** [reply]
I'm assuming you mean "dBm" - dBi is not a unit of power. In that case, yes, you will call `setOutputPower(-5)` and the actual output power will be 21 dBm.

**Reply by farmaceuta777** [reply]
Thank you very much for the answers! I appreciate your help .. :-)
